File: 2025/go/1/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Move struct {
	Dir  string
	Dist int
}

func main() {
	file, err := os.Open("input.dat")
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var l []Move
	
	for scanner.Scan() {
		parts := strings.Fields(scanner.Text())	

		dir := string(parts[0][0])

		move, err := strconv.Atoi(string(parts[0][1:]))
		if err != nil {
			fmt.Println("Error converting move:", err)
			return
		}

		l = append(l, Move{dir, move})
	}
	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	count := 0
	dial := 50
	for _, m := range l {
		if m.Dir == "L" {
			dial = (dial - m.Dist) % 100
		} else if m.Dir == "R" {
			dial = (dial + m.Dist) % 100
		}

		if dial == 0 {
			count++
		}
	}

	fmt.Println(count)
}

File: 2025/python/01/part1.py
raw = open('input.dat').readlines()

count = 0
dial = 50

for line in raw:
    line = line.strip()
    if not line: continue
    move = int(line[1:])
    if line[0] == 'L':
        dial = (dial - move) % 100
    elif line[0] == 'R':
        dial = (dial + move) % 100
    if dial == 0:
        count += 1
        
print(count)
